### PR TITLE
Add gtk2 and libnotify

### LINKED ISSUE #3: Add libnotify
Firefox isn't linked against it but uses it for showing native looking system notifications. Without it they will be not that pretty (although still usable).

cc: @barthalion @valentindavid

### LINKED ISSUE #4: Add gtk2 from shared-modules
`libmozgtk.so` (the one from `gtk2` subfolder) is linked against `libgtk-x11-2.0.so.0` and  `libgdk-x11-2.0.so.0` which come from to gtk2 and aren't available in runtime. AFAIK it's only needed for flash support which is going to be removed at some point. Still adding [gtk2 from shared-modules](https://github.com/flathub/shared-modules/tree/master/gtk2) may be considered for completion.

cc: @barthalion @valentindavid

### PR DESCRIPTION
**Pull request by Erick555**
Fixes https://github.com/flathub/org.mozilla.Firefox.BaseApp/issues/4
Fixes https://github.com/flathub/org.mozilla.Firefox.BaseApp/issues/3

### CODE REVIEW

File: org.mozilla.Firefox.BaseApp.json [hunk 1]
```diff
@@ -3,8 +3,35 @@
     "runtime": "org.freedesktop.Platform",
     "runtime-version": "19.08",
     "sdk": "org.freedesktop.Sdk",
-    "separate-locales": false,
+    "cleanup": [
+        "/include",
+        "/lib/pkgconfig",
+        "/share/pkgconfig",
```
Comment (TingPing) [inline org.mozilla.Firefox.BaseApp.json:9]: How do you expect Firefox to build without these

File: org.mozilla.Firefox.BaseApp.json [hunk 2]
```diff
@@ -3,8 +3,35 @@
...
+        "/share/aclocal",
+        "/share/gtk-doc",
+        "/man",
+        "/share/man",
+        "*.la", "*.a"
```
Comment (TingPing) [inline org.mozilla.Firefox.BaseApp.json:14]: Also half of these are copy-pasted nonsense.

File: org.mozilla.Firefox.BaseApp.json (hunk 1 above)
Comment (Erick555) [reply org.mozilla.Firefox.BaseApp.json:9]: I expect this baseapp won't be used to build firefox but only for packaging official binaries from mozilla.

File: org.mozilla.Firefox.BaseApp.json (hunk 2 above)
Comment (Erick555) [reply org.mozilla.Firefox.BaseApp.json:14]: Possibly but it shouldn't hurt. I wanted to be a future proof completionist :smile: .

Most of those were in [initial commit](https://github.com/flathub/org.mozilla.Firefox.BaseApp/commit/17403b19a79e7e95c3801d39e3421c984c8165b7#diff-2da58ed2a3ebc224d82f5248c8905b72R7), then were removed after switch to shared-modules which handle those on their own. Cleanup is needed again for libnotify and for whatever else will be added in the future.

File: org.mozilla.Firefox.BaseApp.json (hunk 1 above)
Comment (TingPing) [reply org.mozilla.Firefox.BaseApp.json:9]: Lame.

Comment (Erick555) [reply org.mozilla.Firefox.BaseApp.json:9]: From Mozilla POV it's way easier to build things once and then package it to various formats like tar, deb, snap and flatpak.

From my experience, building ff with flatpak requires various quirks and adjustments with every major version update which happens once in couple of weeks unless every dep is vendored including rust and node and that defeats the point of building it exclusively in first place.

I'm afraid that if building in flatpak was requirement then official ff flatpak could never happen.

**Pull request merged**